src/libs/blueprint/conduit_blueprint_mesh_matset_xforms.cpp

```diff
@@ -504,71 +504,165 @@ to_silo(const conduit::Node &field,
 }
 
 //-----------------------------------------------------------------------------
+// venn full -> sparse by element
 void
-multi_buffer_by_element_to_uni_buffer_by_element(const conduit::Node &matset,
-                                                 conduit::Node &dest,
+multi_buffer_by_element_to_uni_buffer_by_element(const conduit::Node &src_matset,
+                                                 const conduit::Node &src_field,
+                                                 conduit::Node &dest_matset,
+                                                 conduit::Node &dest_field,
                                                  const float64 epsilon)
 {
+    dest_matset.reset();
+    dest_field.reset();
+
     // set the topology
-    dest["topology"].set(matset["topology"]);
+    dest_matset["topology"].set(src_matset["topology"]);
+
+    if (src_field.dtype().is_object())
+    {
+        // copy over info from the old field
+        dest_field["association"].set(src_field["association"]);
+        dest_field["topology"].set(src_field["topology"]);
+        dest_field["matset"].set(src_field["matset"]);
```

Comment:
do we need to set this according to the new matset name?

Not sure we can always infere from passed dest_matset, so this might need to be an option?

Reply:
Do you think we should allow users to pass in a new matset name? If we're doing that, why not just let them change the matset name in the result field if they do end up naming their new matset something different?

Reply:
Ok, I'll add this functionality.